onnxruntime/core/framework/kernel_registry.cc

```diff
@@ -233,9 +233,10 @@ Status KernelRegistry::TryCreateKernel(const onnxruntime::Node& node,
                                        const FuncManager& funcs_mgr,
                                        const DataTransferManager& data_transfer_mgr,
                                        /*out*/ std::unique_ptr<OpKernel>& op_kernel) const {
-  const KernelCreateInfo* kernel_create_info;
+  const KernelCreateInfo* kernel_create_info = nullptr;
   ORT_RETURN_IF_ERROR(TryFindKernel(node, execution_provider.Type(), &kernel_create_info));
-
+  ORT_ENFORCE(nullptr != kernel_create_info, "kernel info must not be NULL");
```

Comment:
> ORT_ENFORCE(nullptr != kernel_create_info [](start = 1, length = 42)

I think this must be a simple assert(). It would indicate a bug that TryFindKernel is successful then it must be non null but not require a check since we already check the error code above

Reply:
We are relying on ORT_ENFORCE as "assert" in ORT, no?